index.js

```diff
@@ -0,0 +1,135 @@
+const exec = require('@actions/exec');
+const core = require('@actions/core');
+const github = require('@actions/github');
+const { parse: gitDiffParser } = require('what-the-diff');
+const fs = require('fs-extra');
+const simpleGit = require('simple-git');
+const env = process.env;
+
+function matchExact(r, str) {
+  let match = str.match(r);
+  return match && str === match[0];
+}
+
+function fixNewLineEOF(b) {
+  // If file is empty
+  if (b.length < 1) {
+    return b;
+  }
+  // If file is not empty
+  b = b.replace(/[ \t\n]*$/, '\n');
+
+  if (b.length === 1) {
+    // if the remaining character is a whitespace
+    if (/[ \t\n]*$/.test(b)) {
+      return '';
+    } else {
+      return b + '\n';
+    }
+  }
+  return b;
+}
```

Comment:
Used a simple approach of removing all trailing whitespace and replacing it with a single newline character. Some additional cases are handled when there's only `\n` present in the file.